java/src/main/java/org/rocksdb/ReadOptions.java

```diff
@@ -440,13 +440,12 @@ public ReadOptions setIgnoreRangeDeletions(final boolean ignoreRangeDeletions) {
    * @param iterateLowerBound Slice representing the upper bound
    * @return the reference to the current ReadOptions.
    */
-  public ReadOptions setIterateLowerBound(final Slice iterateLowerBound) {
+  public ReadOptions setIterateLowerBound(final AbstractSlice<?> iterateLowerBound) {
     assert(isOwningHandle());
-    if (iterateLowerBound != null) {
-      // Hold onto a reference so it doesn't get garbage collected out from under us.
-      iterateLowerBoundSlice_ = iterateLowerBound;
-      setIterateLowerBound(nativeHandle_, iterateLowerBoundSlice_.getNativeHandle());
-    }
+    setIterateLowerBound(
```

Comment:
This is a change in behavior that also allows the lower bound to be cleared (by passing in null).  I am not saying that is a bad thing, but a test should probably be added to verify that it doesn't cause any issues.

Reply:
Test added